src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/tests/latest/test_redis_scenario.py

```diff
@@ -13,21 +13,34 @@ def test_redis_cache(self, resource_group):
 
         self.kwargs = {
             'rg': resource_group,
-            'loc': 'WestUS',
+            'loc': 'WestUS2',
             'name': self.create_random_name(prefix='redis', length=24),
-            'sku': 'basic',
-            'size': 'C0',
+            'sku': 'premium',
+            'sku2': 'basic',
+            'size': 'P1',
             'name2': self.create_random_name(prefix='redis', length=24),
-            'size2': 'C1',
-            'settings': "{\\\"hello\\\":1}"
+            'size2': 'C0',
+            'tls_version': '1.2',
+            'tags': "{\\\"test\\\":\\\"tryingzones\\\"}",
+            'zones': "{\\\"1\\\",\\\"2\\\"}",
+            'settings': "{\\\"hello\\\":1}",
+            'schedule_entries': "[{\\\"dayOfWeek\\\":\\\"Monday\\\",\\\"startHourUtc\\\":\\\"00\\\",\\\"maintenanceWindow\\\":\\\"PT5H\\\"}]",
+            'schedule_entries_update': "[{\\\"dayOfWeek\\\":\\\"Tuesday\\\",\\\"startHourUtc\\\":\\\"01\\\",\\\"maintenanceWindow\\\":\\\"PT10H\\\"}]"
         }
 
-        self.cmd('az redis create -n {name} -g {rg} -l {loc} --sku {sku} --vm-size {size}')
+        self.cmd('az redis create -n {name} -g {rg} -l {loc} --sku {sku} --vm-size {size} --minimum-tls-version {tls_version} --tags {tags} --zones {zones}')
 
-        self.cmd('az redis create -n {name2} -g {rg} -l {loc} --sku {sku} --vm-size {size2} --tenant-settings {settings}')
+        self.cmd('az redis create -n {name2} -g {rg} -l {loc} --sku {sku2} --vm-size {size2} --tenant-settings {settings}')
         self.cmd('az redis show -n {name} -g {rg}', checks=[
             self.check('name', '{name}'),
             self.check('provisioningState', 'Creating')
         ])
+        self.cmd('az redis list')
         self.cmd('az redis list -g {rg}')
         self.cmd('az redis list-keys -n {name} -g {rg}')
+        self.cmd('az redis patch-schedule create -n {name} -g {rg} --schedule-entries {schedule_entries}')
+        self.cmd('az redis patch-schedule update -n {name} -g {rg} --schedule-entries {schedule_entries_update}')
+        self.cmd('az redis patch-schedule show -n {name} -g {rg}')
+        self.cmd('az redis patch-schedule delete -n {name} -g {rg}')
+        self.cmd('az redis firewall-rules list -n {name} -g {rg}')
+        self.cmd('az redis server-link list -n {name} -g {rg}')
```

Comment:
It would be good to test az redis server-list set. You should also break up this test into different tests.

Test basic redis commands in one test, then patch-schedule tests in the other, firewall-rules in the other and so on...

Reply:
I have broken down the tests. Some commands can't be tested this way because they depend on redis cache to be created and redis cache creation could take from 20-40 mins